### ISSUE TITLE
Using R Pi 3?

### ISSUE THREAD

**Issue by rdgarlo** [mrworf/serial-bridge#2]
This is a great project - thank you for providing it!

I have the hardware set up and I thought I could use my existing Raspberry Pi 3. It has several USB ports. However, it throws this error:

pi@raspberrypi:~ $ cd /opt/serial-bridge
pi@raspberrypi:/opt/serial-bridge $ python3 bridge.py --debug
  File "bridge.py", line 21
    logging.info(f'Found {serial} port')
                                      ^
SyntaxError: invalid syntax
pi@raspberrypi:/opt/serial-bridge $ 

------------------------
Should I forget about using the RPi3 and get a separate R Pi Zero?

Thanks for any assistance you can provide.
Bob G.

**Comment by mrworf** [comment]
Actually, it looks like you don't have python3 on your Pi. I did a test on my machine and...

```
ha@development:~/projects/serial-bridge$ python3 bridge.py --debug
Debug logging enabled
DEBUG: Found a total of 0 serial ports
ERROR: Too few ports, cannot bridge
ha@development:~/projects/serial-bridge$ python2 bridge.py --debug
  File "bridge.py", line 21
    logging.info(f'Found {serial} port')
                                      ^
SyntaxError: invalid syntax
```
If you run `python3 --version` what do you get?

**Comment by rdgarlo** [comment]
Hmm ... It *seems* to be there. Admittedly, I know little about Python. I have never used it on this R Pi. Here is the shell response:
pi@raspberrypi:~ $ python3 --version
Python 3.5.3
pi@raspberrypi:~ $ 

Here is the Python shell when I run it from the GUI:
Python 3.5.3 (default, Apr  5 2021, 09:00:41) 
[GCC 6.3.0 20170516] on linux
Type "copyright", "credits" or "license()" for more information.
>>> 

I installed bridge.py by running install.sh (after changing its permissions to make it executable). If I ran install.sh more than once, would it cause problems?

**Comment by rdgarlo** [comment]
I got it working!

* I made a "hello world" script, put it in /opt/serial-bridge, and ran it successfully. This told me that the Python interpreter was working and that the permissions were OK.
* I ran each "import" statement manually in the "Thonny" Python IDE without error. Then I ran an "import" statement with a bogus package name and got an error. This told me that I wasn't missing a package or dependency.
* I upgraded Python to the latest version (3.9.7). I got the same syntax error.
* At a loss for what to try next, I built an entirely new Raspbian OS on a new SD card. This came stock with Python 3.7.3.
* When I got this all set up, the bridge.py script started running! Yay!
* Unfortunately, the bridge.service failed to install, so I did that part manually.

Now my network is up and running. I am using the 2413U plug-in modem. Communications seem solid so far. I am also running a Samba file server and an Open VPN server on this R Pi. I suspected a conflict with one of those, but whatever the problem was seems to be gone. Again, thank you for providing these great instructions and this script!

![IMG_20211003_054716](https://user-images.githubusercontent.com/42189665/135754397-b994ad87-c56f-4254-87d1-3958ea8402ac.jpg)
